### PR TITLE
[WIP][Basic] Add LRUCache and LinkedList

### PR DESCRIPTION
**Pull request by aciidb0mb3r**

### CODE REVIEW

**Comment by abertelrud** [comment]
Looks good, though I do think some unit tests should be added before this is merged.

**Comment by aciidb0mb3r** [comment]
@abertelrud Yes, this is incomplete, we also need on disk support for LRU. I should have added WIP in title. I'll do so now.

File: Sources/Basic/LRUCache.swift [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            defer { evict() }
+
+            // Remove the old node.
+            if let oldNode = store[key] {
+                store[key] = nil
+                lruDataList.remove(oldNode)
+            }
+
+            // If value is nil, we're done.
+            guard let value = value else {
```
Comment (alokjha) [inline Sources/Basic/LRUCache.swift]: the nil check of value should be before removing of oldNode.

**Comment by ivanaxel** [comment]
Wip

**Comment by aciidb0mb3r** [comment]
Closing this until I get time to work on this.

**Pull request closed without merging**